<launch>    
    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />
    
    <group ns="robot0">

      <param name="idRobot" value="0"/>
      <param name="nRobots" value="2"/>
      <param name="baseIdx" value="100"/>
      <param name="base_addr" value="192.168.0."/>
      <param name="modality" value="sim"/>
      <remap from="/clock" to="clock"/>

      <!-- segment_eigen parameters -->
      <param name="n_nearest_neighbours" value="3"/>
      <param name="feature_distance_threshold" value="0.5"/>
      <param name="gc_resolution" value="0.15"/>
      <param name="gc_min_cluster_size" value="6"/>
      <param name="segmentation_mode" value="1"/>

      <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rdc_slam)/rviz/test.rviz" />

      <!--- TF -->
      <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 0 0 0 /map /robot0/camera_init 10" />
      <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera" args="0 0 0 0 0 0 /robot0/camera /robot0/base_link 10" />

      <!--- rdc_slam -->    
      <node pkg="rdc_slam" type="imageProjection"    name="imageProjection"    output="screen"/>
      <node pkg="rdc_slam" type="featureAssociation" name="featureAssociation" output="screen"/>
      <node pkg="rdc_slam" type="cslam"     name="cslam"     output="screen"/>
      <node pkg="rdc_slam" type="transformFusion"    name="transformFusion"    output="screen"/>

      <node pkg="rosbag" type="play" name="playbag" args="--clock xx.bag">
        <remap from="/velodyne_points" to="velodyne_points"/>
        <remap from="/gps_3d_pose" to="base_pose_ground_truth"/> 
      </node>

    </group>

    <group ns="robot1">

      <node pkg="rosbag" type="play" name="playbag" args="--clock xx.bag">
        <remap from="/velodyne_points" to="velodyne_points"/>
        <remap from="/gps_3d_pose" to="base_pose_ground_truth"/> 
      </node>
      
    </group>
    
</launch>
